tests/src/testassembler.cpp

```diff
@@ -150,6 +154,26 @@ auto SimpleAssemblersTest(const PreBasis& preBasis) {
       }
     }
 
+    constexpr double tol = 1e-8;
```

Comment:
Can you write here a comment what you are comparing. Something like *check if allo boundary dofs found by hand and obtainLagrangeNodePositions is the same as using forEachBoundaryDOF.

Reply:
But can you come up with a better test isolated from `testassembler.cpp` since this entangles them unnecessarily.

Reply:
I moved it to `testdirichletvalue.cpp` and added the comment